rest/blip_sync.go

```diff
@@ -41,14 +41,35 @@ type blipHandler struct {
 	db *db.Database
 }
 
+type blipHandlerMethod func(*blipHandler, *blip.Message) error
+
+// Wrap blip handler with code that reloads the user object to make sure that
+// it has the latest channel access grants.
+func userBlipHandler(underlyingMethod blipHandlerMethod) blipHandlerMethod {
+
+	wrappedBlipHandler := func(bh *blipHandler, bm *blip.Message) error {
+
```

Comment:
I was going to suggest checking whether db.user is nil before calling ReloadUser, but since ReloadUser already guards for that we could omit for now.  Might be worth revisiting in the followup ticket/fix for code clarity.